### PR TITLE
[EC-213] unify timeouts in tests

### PR DESCRIPTION
**Pull request by rtkaczyk**
Replaces the timeouts in tests with values defined in a single place.

Some remarks:
- ~`Thread.sleep`: arguments replaced - not sure if this is desired~ - some of those were not needed at all, others replaced with predefined values
- ~`time.advance`: mock scheduler uses values defined at call site, instead configured/predefined values - not changed~ - this will be addressed in another PR
- ~test times may have increased (?)~ - fixed

~TODO: try overriding Circle CI inferred build steps (reduntant `sbt test:compile`)~ - done

### CODE REVIEW

**Comment by rtkaczyk** [comment]
All unit tests passed:

```
[info] Run completed in 5 minutes, 15 seconds.
[info] Total number of tests run: 895
[info] Suites: completed 78, aborted 0
[info] Tests: succeeded 895, failed 0, canceled 0, ignored 0, pending 0
```

But there's quite significant increase in the duration - from 3 minutes, 41 seconds in build 2147.

The build failed because of `it tests` - they were not touched in this PR.

**Comment by ntallar** [comment]
It seems `PeerManagerSpec` has it's own `patienceConfig` defined, maybe should we use some of ones defined in `patience.scala`?

File: src/test/scala/io/iohk/ethereum/jsonrpc/JsonRpcControllerSpec.scala [hunk 1]
```diff
@@ -1307,7 +1304,7 @@ class JsonRpcControllerSpec extends FlatSpec with Matchers with PropertyChecks w
       override val blockCacheSize: Int = 30
       override val ommersPoolSize: Int = 30
       override val txPoolSize: Int = 30
-      override val poolingServicesTimeout: FiniteDuration = 3.seconds
+      override val poolingServicesTimeout: FiniteDuration = Timeouts.normalTimeout
     }
 
     val filterConfig = new FilterConfig {
```
Comment (ntallar) [inline src/test/scala/io/iohk/ethereum/jsonrpc/JsonRpcControllerSpec.scala]: Minor detail: Should we use some of the timeouts from `Timeouts` for all the timeouts in this `FilterConfig`?

**Comment by ntallar** [comment]
Should we remove the commented `Thread.sleep` and `time.advance`? It seems to make sense for them to not be needed.

File: src/test/scala/io/iohk/ethereum/blockchain/sync/SyncControllerSpec.scala [hunk 2]
```diff
@@ -662,7 +662,7 @@ class SyncControllerSpec extends FlatSpec with Matchers {
 
     val peer = Peer(new InetSocketAddress("127.0.0.1", 0), peerTestProbe.ref)
 
-    time.advance(1.seconds)
+    //time.advance(1.seconds)
```
Comment (rtkaczyk) [inline src/test/scala/io/iohk/ethereum/blockchain/sync/SyncControllerSpec.scala]: remove

File: src/test/scala/io/iohk/ethereum/jsonrpc/EthServiceSpec.scala [hunk 3]
```diff
@@ -513,7 +512,7 @@ class EthServiceSpec extends FlatSpec with Matchers with ScalaFutures with MockF
     (appStateStorage.getBestBlockNumber _).expects().returning(0)
     ethService.getWork(GetWorkRequest())
 
-    Thread.sleep(1.seconds.toMillis)
+    //Thread.sleep(Timeouts.normalTimeout.toMillis)
```
Comment (rtkaczyk) [inline src/test/scala/io/iohk/ethereum/jsonrpc/EthServiceSpec.scala]: remove

File: src/test/scala/io/iohk/ethereum/jsonrpc/EthServiceSpec.scala [hunk 4]
```diff
@@ -527,7 +526,7 @@ class EthServiceSpec extends FlatSpec with Matchers with ScalaFutures with MockF
     (appStateStorage.getBestBlockNumber _).expects().returning(0)
     ethService.submitWork(SubmitWorkRequest(ByteString("nonce"), ByteString(Hex.decode("01" * 32)), ByteString(Hex.decode("01" * 32))))
 
-    Thread.sleep(1.seconds.toMillis)
+    //Thread.sleep(Timeouts.normalTimeout.toMillis)
```
Comment (rtkaczyk) [inline src/test/scala/io/iohk/ethereum/jsonrpc/EthServiceSpec.scala]: remove

File: src/test/scala/io/iohk/ethereum/jsonrpc/EthServiceSpec.scala [hunk 5]
```diff
@@ -539,7 +538,7 @@ class EthServiceSpec extends FlatSpec with Matchers with ScalaFutures with MockF
 
     ethService.submitHashRate(SubmitHashRateRequest(42, ByteString("id")))
 
-    Thread.sleep(1.seconds.toMillis)
+    //Thread.sleep(Timeouts.normalTimeout.toMillis)
```
Comment (rtkaczyk) [inline src/test/scala/io/iohk/ethereum/jsonrpc/EthServiceSpec.scala]: remove

File: src/main/scala/io/iohk/ethereum/jsonrpc/EthService.scala [hunk 6]
```diff
@@ -187,7 +187,7 @@ class EthService(
 
   lazy val blockchain = BlockchainImpl(blockchainStorages)
 
-  val minerTimeOut: Long = 5.seconds.toMillis
+  val minerTimeOut: Long = 5.seconds.toMillis // TODO: configure timeout
```
Comment (adamsmo) [inline src/main/scala/io/iohk/ethereum/jsonrpc/EthService.scala]: do we have task / separate PR for adding configuration for this?

Comment (rtkaczyk) [inline src/main/scala/io/iohk/ethereum/jsonrpc/EthService.scala]: I'm not aware of it, but it was discussed the other day. It's not the scope of this PR, though, so I imagine another task will be planned ( @mcsherrylabs )

**Pull request merged**